providers/src/airflow/providers/http/hooks/http.py

```diff
@@ -143,6 +151,18 @@ def get_conn(self, headers: dict[Any, Any] | None = None) -> requests.Session:
         if headers:
             session.headers.update(headers)
 
+        if self.adapter:
+            scheme = urlparse(self.base_url).scheme if self.base_url else "https"
+            session.mount(f"{scheme}://", self.adapter)
+        elif self.tcp_keep_alive:
+            keep_alive_adapter = TCPKeepAliveAdapter(
+                idle=self.keep_alive_idle,
+                count=self.keep_alive_count,
+                interval=self.keep_alive_interval,
+            )
+            session.mount("http://", keep_alive_adapter)
+            session.mount("https://", keep_alive_adapter)
```

Comment:
We do we need to instantiate `TCPKeepAliveAdapter` here if we already have it in the `run` method?